README.md

```diff
@@ -29,12 +29,41 @@ A third party security audit was performed by Cure53, you can see the full repor
 
 `runc` only supports Linux. It must be built with Go version 1.21 or higher.
 
+### Pre-Requisites
+
+#### Go
+
 NOTE: if building with Go 1.22.x, make sure to use 1.22.4 or a later version
 (see [issue #4233](https://github.com/opencontainers/runc/issues/4233) for
 more details).
 
-In order to enable seccomp support you will need to install `libseccomp` on your platform.
-> e.g. `libseccomp-devel` for CentOS, or `libseccomp-dev` for Ubuntu
+#### Utilities and Libraries
+
+In addition to Go, building `runc` requires multiple utilities and libraries to be installed on your system.
+
+On Ubuntu/Debian, you can install the required dependencies with:
+
+```bash
+apt update && apt install -y make gcc libseccomp-dev pkg-config git
+```
+
+On CentOS/Fedora, you can install the required dependencies with:
+
+```bash
+yum install -y make gcc libseccomp-devel pkg-config git
+```
+
+On Alpine Linux, you can install the required dependencies with:
+
+```bash
+apk --update add make gcc libseccomp-dev musl-dev linux-headers git
```

Comment:
Why is alpine the only one with linux.headers dep?

Reply:
I don't actually know. Either I missed something, or it needs to add deps on Alpine which are there already on the others. That latter doesn't sound right to me, but I will run a clean test.

Reply:
Ran clean tests.

* `alpine:3.20`: I missed `bash`, added that to the doc. It builds cleanly now, both `make runc` and `make static`
* `ubuntu:22.04`:  I didn't get as far as the headers question, because gcc errors out with `gcc: error: unrecognized command-line option '-m64'`. Running gcc 11.4.0.
* `fedora:40`: same as ubuntu. I didn't get as far as the headers question, because gcc errors out with `gcc: error: unrecognized command-line option '-m64'`. Running gcc 14.1.1

In all cases, running go 1.22.5.

I _did_ try inside the golang container from the `Dockerfile` in this repo: `golang:1.22-bookworm`. I do not get the same gcc errors. In there, it is gcc 12.2.0.

What obvious thing am I missing?

Reply:
Are you cross-compiling? It seems like https://github.com/opencontainers/runc/blob/main/cc_platform.mk is causing issues again.

Does it work if you comment its include in the Makefile?

Reply:
> Are you cross-compiling? 

Not intentionally. 😄 

> Does it work if you comment its include in the Makefile?

Same error:

```sh
root@df3feb358386:/pwd# make static
rm -f libcontainer/dmz/binary/runc-dmz
go generate -tags "seccomp urfave_cli_no_docs " ./libcontainer/dmz
make[1]: Entering directory '/pwd/libcontainer/dmz'
gcc -fno-asynchronous-unwind-tables -fno-ident -s -Os -nostdlib -lgcc -static -o binary/runc-dmz _dmz.c
strip -gs binary/runc-dmz
make[1]: Leaving directory '/pwd/libcontainer/dmz'
go build -trimpath -buildmode=pie  -tags "seccomp urfave_cli_no_docs  netgo osusergo" -ldflags "-X main.gitCommit=v1.2.0-rc.2-22-g419ef873 -X main.version=1.2.0-rc.2+dev -linkmode external -extldflags -static-pie " -o runc .
# runtime/cgo
gcc: error: unrecognized command-line option '-m64'
make: *** [Makefile:105: static-bin] Error 1
```

Or if I do it explicitly:

```sh
root@df3feb358386:/pwd# make -n static
rm -f libcontainer/dmz/binary/runc-dmz
go generate -tags "seccomp urfave_cli_no_docs " ./libcontainer/dmz
go build -trimpath -buildmode=pie  -tags "seccomp urfave_cli_no_docs  netgo osusergo" -ldflags "-X main.gitCommit=v1.2.0-rc.2-22-g419ef873 -X main.version=1.2.0-rc.2+dev -linkmode external -extldflags -static-pie " -o runc .
if test -s libcontainer/dmz/binary/runc-dmz; then \
        set -Eeuo pipefail; \
        export LC_ALL=C; \
        diff -u \
                <(readelf -h runc | grep -E "(Machine|Flags):") \
                <(readelf -h libcontainer/dmz/binary/runc-dmz | grep -E "(Machine|Flags):"); \
fi


root@df3feb358386:/pwd# go build -trimpath -buildmode=pie  -tags "seccomp urfave_cli_no_docs  netgo osusergo" -ldflags "-X main.gitCommit=v1.2.0-rc.2-22-g419ef873 -X main.version=1.2.0-rc.2+dev -linkmode external -extldflags -static-pie " -o runc .
# runtime/cgo
gcc: error: unrecognized command-line option '-m64'
```

Reply:
The `golang:1.22-bookworm`, which does not generate the error, has the same command-line. 🤷‍♂️